lib/API/DX/Device.cpp

```diff
@@ -244,9 +244,12 @@ class DXDevice : public offloadtest::Device {
     ComPtr<ID3D12Resource> Upload;
     ComPtr<ID3D12Resource> Buffer;
     ComPtr<ID3D12Resource> Readback;
+    // In unmapped cases, the Heap lifetime needs to be preserved
```

Comment:
I don't understand what this comment means.

Reply:
Although I'm not trying to push you too hard to a "always use reserved resources" design, this is an example where using the same type of resource in both cases would lead to a simpler implementation.